File: Data Structures & Algorithms/move-zeroes/submission-1.java
class Solution {
    public void moveZeroes(int[] nums) {
        int l = 0;
        int r = 0;
        while(r < nums.length){
            if(nums[r] == 0){
                r++;
            }else{
                nums[l] = nums[r];
                r++;
                l++;
            }
        }
        while(l < nums.length){
            nums[l] = 0;
            l++;
        }
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        combo = []
        comboSum = 0
        def backtrack(i):
            nonlocal res, combo, comboSum
            if i >= len(nums) or comboSum > target:
                if comboSum == target:
                    res.append(combo[:])
                return
            comboSum += nums[i]
            combo.append(nums[i])
            backtrack(i)
            combo.pop()
            comboSum -= nums[i]
            backtrack(i+1)
        
        backtrack(0)

        return res
            